cdap-authenticator-ext-gcp/src/main/java/io/cdap/cdap/authenticator/gcp/GCPRemoteAuthenticator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.authenticator.gcp;
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.proto.security.Credential;
+import io.cdap.cdap.security.spi.authenticator.RemoteAuthenticator;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.time.Clock;
+import java.time.Instant;
+import javax.annotation.Nullable;
+
+/**
+ * {@link RemoteAuthenticator} Authenticator which returns Google application default credentials.
+ * For additional information, see https://google.aip.dev/auth/4110.
+ */
+public class GCPRemoteAuthenticator implements RemoteAuthenticator {
```

Comment:
how is this different than `GceRemoteAuthenticator`? Do we need both?

Reply:
`GceRemoteAuthenticator` explicitly reads only from the GCE internal metadata server (and thus does not support other application default credentials such as the `GOOGLE_APPLICATION_CREDENTIALS` environment variable). Theoretically we do not need `GceRemoteAuthenticator`, but my understanding is that it is more lightweight because it is baked into app-fabric and does not require the Google client libraries.